libs/extensions/include/NanoDimuonVertex.hpp

```diff
@@ -15,31 +15,58 @@ typedef Collection<std::shared_ptr<NanoDimuonVertex>> NanoMuonVertices;
 
 class NanoDimuonVertex {
  public:
-  NanoDimuonVertex(std::shared_ptr<PhysicsObject> physicsObject_);
+  NanoDimuonVertex(std::shared_ptr<PhysicsObject> physicsObject_, const std::shared_ptr<Event> event);
 
   auto Get(std::string branchName) { return physicsObject->Get(branchName); }
   float GetAsFloat(std::string branchName) { return physicsObject->GetAsFloat(branchName); }
   std::string GetOriginalCollection() { return physicsObject->GetOriginalCollection(); }
   void Reset() { physicsObject->Reset(); }
 
+  std::shared_ptr<PhysicsObject> Muon1() {return muon1;}
+  std::shared_ptr<PhysicsObject> Muon2() {return muon2;}
+
   bool isDSAMuon1() { return float(physicsObject->Get("isDSAMuon1")) == 1.; }
   bool isDSAMuon2() { return float(physicsObject->Get("isDSAMuon2")) == 1.; }
-  float muonIndex1() { return float(physicsObject->Get("idx1")); }
-  float muonIndex2() { return float(physicsObject->Get("idx2")); }
+  float muonIndex1() { return float(physicsObject->Get("originalMuonIdx1")); }
+  float muonIndex2() { return float(physicsObject->Get("originalMuonIdx2")); }
+  bool isValid() { return (float)Get("isValid") > 0; }
 
   std::shared_ptr<PhysicsObject> GetPhysicsObject() { return physicsObject; }
 
   std::string GetVertexCategory();
   std::pair<std::shared_ptr<PhysicsObject>,std::shared_ptr<PhysicsObject>> GetMuons(const std::shared_ptr<Event> event);
-  float GetDimuonChargeProduct(const std::shared_ptr<Event> event);
 
+  TLorentzVector GetFourVector();
+  float GetInvariantMass() { return GetFourVector().M(); }
+  float GetPt() { return GetFourVector().Pt(); }
+  float GetEta() { return GetFourVector().Eta(); }
+  float GetPhi() { return GetFourVector().Phi(); }
+
+  TVector3 GetLxyzFromPV() { return Lxyz; };
+  float GetLxyFromPV() { return Lxyz.Perp(); }
+
+  float GetCollinearityAngle();
+  float GetDeltaPixelHits();
+
+  float GetDimuonChargeProduct();
+
+  bool PassesDCACut();
   bool PassesChi2Cut();
+  bool PassesCollinearityAngleCut();
+  bool PassesDeltaPixelHitsCut();
   bool PassesMaxDeltaRCut();
   bool PassesMinDeltaRCut();
-  bool PassesDimuonChargeCut(const std::shared_ptr<Event> event);
+  bool PassesLxyCut();
+  bool PassesDimuonChargeCut();
+  bool PassesVxySigmaCut();
+  bool PassesHitsBeforeVertexCut();
```

Comment:
I feel like these things don't really belong here. Imagine that someone from a different analysis wants to use this class and they see a method `PassesDCACut` - but what's the value of this cut? Does it make sense for the other analysis?

Perhaps it would be fine if all these methods took arguments specifying the value of the cut, but then I'd rather use methods like `GetDCA`, `GetChi2`, etc. somewhere else (in some selections manager) I check if these values are within some ranges (specified in the config).

Reply:
It is actually using the std::map<std::string, float> muonVertexCuts and I've just specified strings that is read from a config file. But I also agree, once I started to add more cuts I think it got more and more analysis specific.

Reply:
I removed it now, it's handled in tea_ttalps now